### PR TITLE
Add email notification for JOB_RUNNING_TO_FINISHED_WITH_ERRORS

### PR DESCRIPTION
**Pull request by mboussaa**
Add email notification for jobs having the notification event `JOB RUNNING TO FINISHED WITH ERRORS`:

-Consider a job `running to finished with errors` if it has one or more `faulty` tasks
-Add event to SchedulerEvent (`JOB_RUNNING_TO_FINISHED_WITH_ERRORS` or `Job running to finished with errors` could be introduced as GI)
-For now, the filtering concerns only past finished jobs, which means jobs having finished as status.

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobEmailNotification.java [hunk 1]
```diff
@@ -125,50 +126,83 @@ public boolean doCheckAndSend(boolean withAttachment)
...
                                                     "Error sending email: " + e.getMessage(),
                                                     e);
         }
     }
 
+    private boolean hasFaultyTasks() {
+        List<TaskState> tasks = jobState.getTasks();
+        boolean withErrors = tasks.stream()
+                                  .map(task -> task.getStatus().name().toLowerCase())
+                                  .anyMatch(t -> t.equals("Faulty".toLowerCase()));
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobEmailNotification.java]: why not using JobInfo contained in notification data?

Besides this, we are interested in getNumberOfFailedTasks, getNumberOfFaultyTasks and getNumberOfInErrorTasks

Comment (mboussaa) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobEmailNotification.java]: @fviale thanks for notifying this. The new commit should have fixed that.

**Comment by fviale** [comment]
as a matter of fact, maybe it could be useful to add a unit test for the new feature you introduced

File: scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/core/JobEmailNotificationTest.java [hunk 2]
```diff
@@ -302,6 +302,26 @@ public void testFinished() throws Exception {
         verifyNoMoreInteractions(stubbedSender);
     }
 
+    @Test
+    public void testFinishedWithErrors() throws Exception {
```
Comment (fviale) [inline scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/core/JobEmailNotificationTest.java]: you should add a test to verify that, if a job contains no error, with NOTIFICATION_EVENTS sets the same as this test, no notification should occur.

**Pull request merged**